debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/SinkRecordDescriptor.java

```diff
@@ -154,6 +157,9 @@ public Struct getKeyStruct(PrimaryKeyMode primaryKeyMode) {
     }
 
     public Struct getAfterStruct() {
+        if (isTombstone()) {
+            return ((Struct) record.key());
```

Comment:
Doing this here could lead to potentially issues in the non-tombstone cases, don't you think?

Reply:
I can see your concern this is generic code that is used in other places, though if this is a non-tombstone event this `if` condition will be skipped and the original path will be executed. If this is a tombstone then the there is no value struct as that's part of the isTombstone test so we would without this code return null.
I also checked and there's no after structure when we are dealing with a key. I think this is safe from my testing

Reply:
this is no longer required